packages/app/components/strategies/StrategyCard.tsx

```diff
@@ -0,0 +1,99 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import { cx } from "class-variance-authority";
+
+import { Button, CaptionText, Icon } from "@/ui";
+import { Strategy, useStrategyContext } from "@/contexts";
+import { TokenLogoPair } from "@/components/TokenLogoPair";
+
+import { FREQUENCY_LABEL } from "./constants";
+
+interface StrategyCardProps {
+  strategy: Strategy;
+}
+
+export const StrategyCard = ({ strategy }: StrategyCardProps) => {
+  const [isHovered, setIsHovered] = useState(false);
+  const [isSelected, setIsSelected] = useState(false);
+  const { selectedStrategy, setSelectedStrategy } = useStrategyContext();
+
+  const { buyToken, sellToken } = strategy;
+
+  const strategyTotalDetails = [
+    {
+      label: "Total Amount",
+      totalAmount: `${strategy.totalSellAmount} ${sellToken.symbol}`,
+    },
+    { label: "No. of days", totalAmount: strategy.daysAmount },
+  ];
+
+  useEffect(() => {
+    setIsSelected(selectedStrategy?.id === strategy.id);
+  }, [selectedStrategy, strategy.id]);
```

Comment:
I think `const isSelected = selectedStrategy?.id === strategy.id;` works, every time `selectedStrategy` changes there is a re-render.

Reply:
You're right lol, good catch! Thanks :D